.github/workflows/publish.yml

```diff
@@ -20,7 +20,10 @@ name: Publish
 on:
   push:
     tags:
-      - "*"
+      # Trigger this workflow when tag follows the versioning format: v<major>.<minor>.<patch> OR v<major>.<minor>.<patch>-rc.<release_candidate>
+      # Example valid tags: v0.4.0, v0.4.0-rc.1
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc.[0-9]+"
```

Comment:
I consider myself a bit of a regex connoisseur.

```suggestion
      - "v\d+.\d+.\d+"
      - "v\d+.\d+.\d+-rc.\d+"
```

<img width="1139" alt="image" src="https://github.com/user-attachments/assets/967eec62-410b-410a-a20e-f4a2f5bb4aca" />